settlement/dymension/dymension.go

```diff
@@ -212,76 +212,96 @@ func (d *HubClient) PostBatch(batch *types.Batch, daClient da.Client, daResult *
-	subscription, err := d.pubsub.Subscribe(d.ctx, postBatchSubscriberClient, settlement.EventQueryNewSettlementBatchAccepted)
+	subscription, err := d.pubsub.Subscribe(d.ctx, postBatchSubscriberClient, settlement.EventQueryNewSettlementBatchAccepted, 100)
 	if err != nil {
 		return fmt.Errorf("pub sub subscribe to settlement state updates: %w", err)
 	}
 
 	//nolint:errcheck
 	defer d.pubsub.Unsubscribe(d.ctx, postBatchSubscriberClient, settlement.EventQueryNewSettlementBatchAccepted)
 
-	// Try submitting the batch to the settlement layer. If submission (i.e. only submission, not acceptance) fails we emit an unhealthy event
-	// and try again in the next loop. If submission succeeds we wait for the batch to be accepted by the settlement layer.
-	// If it is not accepted we emit an unhealthy event and start again the submission loop.
+	// Try submitting the batch to the settlement layer:
+	// 1.broadcasting the transaction to the blockchain (with retries).
+	// 2.waiting for the batch to be accepted by the settlement layer.
 	for {
-		select {
-		case <-d.ctx.Done():
-			return d.ctx.Err()
-		default:
-			err := d.submitBatch(msgUpdateState)
-			if err != nil {
-				d.logger.Error(
-					"Submit batch",
-					"startHeight",
-					batch.StartHeight,
-					"endHeight",
-					batch.EndHeight,
-					"error",
-					err,
-				)
-
-				// Sleep to allow context cancellation to take effect before retrying
-				time.Sleep(100 * time.Millisecond)
-				continue
+		for {
+			select {
+			case <-d.ctx.Done():
+				return d.ctx.Err()
+			default:
+				err := d.broadcastBatch(msgUpdateState)
+				if err != nil {
+					d.logger.Error(
+						"Submit batch",
+						"startHeight",
+						batch.StartHeight,
+						"endHeight",
+						batch.EndHeight,
+						"error",
+						err,
+					)
+					// If submission (i.e. only submission, not acceptance) fails we try submitting again.
+					// Sleep to allow context cancellation to take effect before retrying
+					time.Sleep(100 * time.Millisecond)
+					continue
+				}
 			}
+			break
 		}
 
 		// Batch was submitted successfully. Wait for it to be accepted by the settlement layer.
 		timer := time.NewTimer(d.batchAcceptanceTimeout)
 		defer timer.Stop()
-
-		select {
-		case <-d.ctx.Done():
-			return d.ctx.Err()
-
-		case <-subscription.Cancelled():
-			return fmt.Errorf("subscription cancelled")
-
-		case <-subscription.Out():
-			d.logger.Debug("Batch accepted", "startHeight", batch.StartHeight, "endHeight", batch.EndHeight)
-			return nil
-
-		case <-timer.C:
-			// Check if the batch was accepted by the settlement layer, and we've just missed the event.
-			includedBatch, err := d.waitForBatchInclusion(batch.StartHeight)
-			if err != nil {
-				d.logger.Error(
-					"Wait for batch inclusion",
-					"startHeight",
-					batch.StartHeight,
-					"endHeight",
-					batch.EndHeight,
-					"error",
-					err,
-				)
-
-				timer.Stop() // we don't forget to clean up
-				continue
+		batchAcceptanceAttempts := 5
```

Comment:
think it's cleaner to have the `5` defined as a global const.

Reply:
i'll add it in this PR
https://github.com/dymensionxyz/dymint/pull/863